src/utils/idParser.js

```diff
@@ -8,8 +8,12 @@ class IdParser {
     static parse(id) {
         let id1;
         let sourceAssigningAuthority;
-
-        const idParts = id.split('|');
+        let idParts = [];
+        if (id && typeof id === 'string') {
+            idParts = id.split('|');
+        } else {
```

Comment:
We should throw validation error if id is not a string

Reply:
Added throw of NotValidatedError. If this isn't the right error to throw, let me know.

Reply:
Instead of throwing an error, We can pass a field in queryRewriterManager.rewriteArgsAsync to enable/disbale the `patientProxyQueryRewriter` and then only enable it for Searches and update the code accordingly. 
This will not only correct the sentry error but also improve the performance of fhir-server.
Reference code: (https://github.com/icanbwell/fhir-server/blob/3.2.34/src/operations/fhirOperationsManager.js#L276)
Also please check if is there any need to call the proxy-patient expander in Graphql and disable it accordingly.
** Currently all the arg-rewriters are called in https://github.com/icanbwell/fhir-server/blob/3.2.34/src/queryRewriters/queryRewriterManager.js